lib/matplotlib/collections.py

```diff
@@ -796,23 +798,29 @@ def _get_default_edgecolor(self):
         # This may be overridden in a subclass.
         return mpl.rcParams['patch.edgecolor']
 
+    def get_hatchcolor(self):
+        return self._hatchcolors
+
     def _set_edgecolor(self, c):
-        set_hatch_color = True
         if c is None:
             if (mpl.rcParams['patch.force_edgecolor']
                     or self._edge_default
                     or cbook._str_equal(self._original_facecolor, 'none')):
                 c = self._get_default_edgecolor()
             else:
                 c = 'none'
-                set_hatch_color = False
         if cbook._str_lower_equal(c, 'face'):
             self._edgecolors = 'face'
+            if self._original_hatchcolor is None:
+                self._set_hatchcolor('black')
```

Comment:
I agree that it shouldn't use hard coded values like 'black'. As mentioned, the `hatch.color` rcparam is changed to `inherit`. So, instead of that, it could fallback to using `patch.edgecolor`. This will occur only if hatchcolor is not set, and edgecolor is set but not valid, like 'none' or 'face'. So, using the default_edgecolor makes sense.